ios/src/TGMapViewController.mm

```diff
@@ -546,10 +549,11 @@ - (void)respondToShoveGesture:(UIPanGestureRecognizer *)shoveRecognizer
     // don't trigger shove on single touch gesture
     if ([shoveRecognizer numberOfTouches] == 2) {
         if (self.gestureDelegate && [self.gestureDelegate respondsToSelector:@selector(recognizer:didRecognizeShoveGesture:)]) {
-            [self.gestureDelegate mapView:self recognizer:shoveRecognizer didRecognizeShoveGesture:displacement];
-        } else {
-            self.map->handleShoveGesture(displacement.y);
+            if (![self.gestureDelegate mapView:self recognizer:shoveRecognizer didRecognizeShoveGesture:displacement]) {
```

Comment:
Should this one be out of the test `if ([shoveRecognizer numberOfTouches] == 2) {`? Implementation developer might still want to decide on that.